### PR TITLE
New docs

### PR DESCRIPTION
**Pull request by HamedFathi**
### Getting Started

* Bindable Setter

### Advanced Scenarios

* CSS-in-JS with Emotion
* TailwindCSS Integration
* Strongly-typed Template
* Write Custom Plugin
* DOM Style Injection

### CODE REVIEW

**Comment by AureliaEffect** [comment]
The benchmarking comparison can be found here: https://au2-bench-viewer.azurewebsites.net/compare?branch=master&commit=0a2d77e206923bcabe43486a3d847f51569b9663&branch=new-docs&commit=827dc3ad5b2b045585a021beb99e39861321991c

**Comment by AureliaEffect** [comment]
The benchmarking comparison can be found here: https://au2-bench-viewer.azurewebsites.net/compare?branch=master&commit=0a2d77e206923bcabe43486a3d847f51569b9663&branch=new-docs&commit=522590a688080450db1095d5fff688dd3b617e86

**Comment by AureliaEffect** [comment]
The benchmarking comparison can be found here: https://au2-bench-viewer.azurewebsites.net/compare?branch=master&commit=0a2d77e206923bcabe43486a3d847f51569b9663&branch=new-docs&commit=b79ba4874b6c40bc82e6e79ef8ffadd004c6750e

File: docs/user-docs/advanced-scenarios/css-in-js-with-emotion.md [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+    }
+}
+```
+
+Surely there are questions about the above code let me answer them one by one.
+
+**What is isInShadow?**
+This method helps us to find out our HTMLElement is inside a shadow-root or not.
+
+**Why shadow-root matter?**
```
Comment (Vheissu) [inline docs/user-docs/advanced-scenarios/css-in-js-with-emotion.md:46]: This should read, "Why does shadow-root matter?"

File: docs/user-docs/advanced-scenarios/css-in-js-with-emotion.md [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+    private isInShadow(element: Element): boolean {
+        return element.getRootNode() instanceof ShadowRoot;
+    }
+}
+```
+
+Surely there are questions about the above code let me answer them one by one.
+
+**What is isInShadow?**
+This method helps us to find out our HTMLElement is inside a shadow-root or not.
```
Comment (Vheissu) [inline docs/user-docs/advanced-scenarios/css-in-js-with-emotion.md:44]: Should read, "This method helps us to find out if our HTMLElement is inside of a shadow-root or not.

**Review by Vheissu — CHANGES_REQUESTED** [review]

**Comment by AureliaEffect** [comment]
The benchmarking comparison can be found here: https://au2-bench-viewer.azurewebsites.net/compare?branch=master&commit=0a2d77e206923bcabe43486a3d847f51569b9663&branch=new-docs&commit=84e2ce81b397b9407bf89843a0d7ac1b9c922077

**Comment by AureliaEffect** [comment]
The benchmarking comparison can be found here: https://au2-bench-viewer.azurewebsites.net/compare?branch=master&commit=ec47fe936caf13cc385d4039ef7f09b2bbce1fd4&branch=new-docs&commit=69f84598bfc72940ef6b6503831b49b3da6166fd

**Review by Vheissu — APPROVED** [review]
This all looks great. Let's get it in.

**Comment by HamedFathi** [comment]
@fkleuver 
Seems ready to merge. If you have any comments let me know.
Thanks.

**Pull request merged**